solutions/CSES Increasing Subsequence II.mdx

```diff
@@ -2,10 +2,24 @@
 id: cses-1748
 source: CSES
 title: Increasing Subsequence II
-author: Benjamin Qi, Óscar Garries
+author: Benjamin Qi, Óscar Garries and Timothy Gao
 ---
 
-Coordinate compress, then range sum.
+Let $dp[i]$ be the number of increasing subsequences ending on the right at $i$.
+Notice that $dp[i]$ can be calculated as the summation of all $dp[j]$ such that $j$ is in the range $[0,i]$ and $nums[j] < nums[i]$.
+
+Naively, this gives an $\mathcal O(N^2)$ solution. In order to speed this up, we can
+coordinate compress, then range sum.
+
+More specifically, we sort all numbers in $nums$ and map each distinct number to its index in the sorted array. This way, we effectively coordinate compress the numbers. 
+
+With this, we can create a BIT, which intially starts with all 0s. As we run through $nums$ from left to right, $dp[i]$ would be summation of all $j<k$ in the BIT, where $k$ is the index $nums[i]$ maps to. Then, we update $k$ in the BIT as preparation for the next query by another number. 
```

Comment:
Link to BIT module [https://usaco.guide/gold/PURS](https://usaco.guide/gold/PURS)